### PR TITLE
Add formatter and corresponding ci

### PR DESCRIPTION
**Pull request by JonasKellerer**
Part of #27

### CODE REVIEW

File: src/example.test.cpp [hunk 1]
```diff
@@ -1,18 +1,15 @@
...
 class ClassToTest {
-   public:
-   virtual void functionToMock() {
-   }
-   virtual void callMockedFunction() {
-      functionToMock();
-   }
+  public:
+   virtual void functionToMock() {}
+   virtual void callMockedFunction() { functionToMock(); }
```
Comment (fengelniederhammer) [inline src/example.test.cpp:8]: Maybe not so important here, but do we really want to allow method declaration and implementation on the same line?

File: src/query_engine/query_engine.cpp [hunk 2]
```diff
@@ -86,18 +102,20 @@ std::unique_ptr<BoolExpression> parse_expression(const Database& db, const rapid
...
       unsigned pos = js["position"].GetUint();
       char ref_symbol = db.global_reference[0].at(pos);
       /// this <= is correct! the negation would flip the exact bit from -1 to +1 and vice versa
-      if (exact <= 0) { /// NucEqEx
-         return std::make_unique<NegEx>(std::make_unique<NucEqEx>(pos, silo::to_symbol(ref_symbol)));
-      } else { /// NucMbEx
-         return std::make_unique<NegEx>(std::make_unique<NucMbEx>(pos, silo::to_symbol(ref_symbol)));
+      if (exact <= 0) {  /// NucEqEx
+         return std::make_unique<NegEx>(std::make_unique<NucEqEx>(pos, silo::to_symbol(ref_symbol))
+         );
```
Comment (fengelniederhammer) [inline src/query_engine/query_engine.cpp:115]: This looks odd. Can we somehow make it look like this?
```suggestion
         return std::make_unique<NegEx>(
            std::make_unique<NucEqEx>(pos, silo::to_symbol(ref_symbol))
         );
```

File: include/silo/database.h [hunk 3]
```diff
@@ -64,30 +64,28 @@ class DatabasePartition {
...
 
    void finalizeBuild(const Dictionary& dict);
 };
 
 struct PreprocessingConfig;
 
 class Database {
-   public:
-   const std::string wd; // working directory
+  public:
```
Comment (fengelniederhammer) [inline include/silo/database.h:80]: Do we really want to have only two spaces indent here?

**Review by fengelniederhammer — APPROVED** [review]

File: src/example.test.cpp (hunk 1 above)
Comment (JonasKellerer) [reply src/example.test.cpp:8]: In general, yes. But we want to have the declaration in the header files and the implementation in the cpp files, where this is done automatically. Thus this should not be an issue.
The example.test.cpp can be deleted later on, when we dont need it anymore for easy reference.

File: src/query_engine/query_engine.cpp (hunk 2 above)
Comment (JonasKellerer) [reply src/query_engine/query_engine.cpp:115]: Good idea. However, with the clang-tidy pullrequest, this should not occur at this point due to renaming.

**Pull request merged**